services/slam/builtin/builtin.go

```diff
@@ -479,19 +483,19 @@ func NewBuiltIn(ctx context.Context, r robot.Robot, config config.Service, logge
 	var dataRate int
 	if svcConfig.DataRateMs == 0 {
 		dataRate = defaultDataRateMs
+		logger.Debugf("no data_rate_msec given, setting to default of %d", defaultDataRateMs)
 	} else {
 		dataRate = svcConfig.DataRateMs
 	}
 
 	var mapRate int
-	if svcConfig.MapRateSec <= 0 {
-		if svcConfig.MapRateSec == -1 {
-			mapRate = 0
-		} else {
-			mapRate = defaultMapRateSec
-		}
+	if svcConfig.MapRateSec == nil {
+		logger.Debugf("no map_rate_secs given, setting to default of %d", defaultMapRateSec)
+		mapRate = defaultMapRateSec
+	} else if *svcConfig.MapRateSec == 0 {
+		logger.Info("setting slam system to localization mode")
```

Comment:
[opt] I know it is redundant, since `mapRate` is `=0` by default (right?) - but you could add a line `mapRate = 0` here to make reading the code easier.